text/0000-more-api-documentation-conventions.md

```diff
@@ -0,0 +1,724 @@
+///
+/// assert_eq!(&[], slice);
+/// ```
+pub fn ref_slice<T>(opt: &Option<T>) -> &[T] {
+    unimplemented!()
+}
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+It’s possible that RFC 505 went far enough, and something this detailed is inappropriate.
+
+# Alternatives
+[alternatives]: #alternatives
+
+We could stick with the more minimal conventions of the previous RFC.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+None.
+
+# Appendix A: Full conventions text
+
+Below is a combination of RFC 505 + this RFC’s modifications, for convenience.
+
+### Summary sentence
+[summary-sentence]: #summary-sentence
+
+In API documentation, the first line should be a single-line short sentence
+providing a summary of the code. This line is used as a summary description
+throughout Rustdoc’s output, so it’s a good idea to keep it short.
+
+The summary line should be written in third person singular present indicative
+form. Basically, this means write ‘Returns’ instead of ‘Return’.
+
+### English
+[english]: #english
+
+This section applies to `rustc` and the standard library.
+
+All documentation for the standard library is standardized on American English,
+with regards to spelling, grammar, and punctuation conventions. Language
+changes over time, so this doesn’t mean that there is always a correct answer
+to every grammar question, but there is often some kind of formal consensus.
+
+One specific rule that comes up often: when quoting something for emphasis,
+use a single quote, and put punctuation outside the quotes, ‘this’. When
+quoting something at length, “use double quotes and put the punctuation
+inside of the quote.” Most documentation will end up using single quotes,
+so if you’re not sure, just stick with them.
+
+### Use line comments
+[use-line-comments]: #use-line-comments
+
+Avoid block comments. Use line comments instead:
+
+```rust
+// Wait for the main task to return, and set the process error code
+// appropriately.
+```
+
+Instead of:
+
+```rust
+/*
+ * Wait for the main task to return, and set the process error code
+ * appropriately.
+ */
+```
+
+Only use inner doc comments `//!` to write crate and module-level documentation,
+nothing else. When using `mod` blocks, prefer `///` outside of the block:
+
+```rust
+/// This module contains tests
+mod test {
+    // ...
+}
+```
+
+over
+
+```rust
+mod test {
+    //! This module contains tests
+
+    // ...
+}
+```
+
+### Using Markdown
+[using-markdown]: #using-markdown
+
+Within doc comments, use Markdown to format your documentation.
+
+Use top level headings # to indicate sections within your comment. Common headings:
+
+* Examples
+* Panics
+* Errors
+* Safety
+* Aborts
+* Undefined Behavior
+
+Even if you only include one example, use the plural form: ‘Examples’ rather
+than ‘Example’. Future tooling is easier this way.
+
+Use backticks (`) to denote a code fragment within a sentence.
+
+Use backticks (```) to write longer examples, like this:
```

Comment:
Not sure to understand what you mean here.... A one backtick is for "highlighting" a "small" element whereas triple backticks are for a whole code block.